doc/ref/gappkg.xml

```diff
@@ -173,5 +184,2074 @@ users.
 
 </Section>
 
-</Chapter>
+<Section Label="Guidelines for Writing a GAP Package">
+<Heading>Guidelines for Writing a GAP Package</Heading>
+
+The remaining part of this chapter explains the basics 
+of how to write a &GAP; package so that it integrates properly into &GAP;.
+<P/>
+
+There are two basic aspects of creating a &GAP; package. 
+<P/> 
+
+First, it is a convenient possibility to load additional functionality into
+&GAP; including a smooth integration of the package documentation. Second,
+a package is a way to make your code available to other &GAP; users.
+<P/>
+
+Moreover, the &GAP; Group may provide some help with redistributing your
+package via the &GAP; website after checking if the
+package provides some new or improved functionality which looks interesting
+for other users, if it contains reasonable documentation, and if it seems
+to work smoothly with the &GAP; library and other distributed packages. In
+this case the package can take part in the &GAP; distribution update
+mechanism and becomes a <E>deposited</E> package.
+<P/>
+
+Furthermore, package authors are encouraged to check if the package would
+be appropriate for the refereeing process and <E>submit</E> it. If the
+refereeing has been successful, the package becomes an <E>accepted</E> package.
+Check out <URL>https://www.gap-system.org/Packages/Authors/authors.html</URL>
+on the &GAP; website for more details.
+<P/>
+
+Below we start with a description how the directory structure of a
+&GAP; package should be constructed and then add remarks on certain aspects
+of creating a package, some of these only apply to some packages. Finally,
+we provide guidelines for the release preparation and its distribution.
+<P/>
+
+</Section>
+
+<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% -->
+<Section Label="Structure of a GAP Package">
+<Heading>Structure of a GAP Package</Heading>
+
+<Index Subkey="for a GAP package">home directory</Index>
+A &GAP; package should have an alphanumeric name (<A>package-name</A>, say);
+mixed case is fine, but there should  be  no  whitespaces. 
```

Comment:
I've made consistent use of `packagename` and `packagedir` in the text instead of `package-name` and `package-dir`. Before, `package-name` was meant to be a variable holding the package name, and not an example of a package name itself, but that's too subtle for the manual :)